payjoin-cli/src/app.rs

```diff
@@ -429,27 +423,21 @@ impl App {
 
     #[cfg(not(feature = "v2"))]
     fn handle_get_bip21(&self, amount: Option<Amount>) -> Result<Response<Body>, Error> {
+        use payjoin::{PayjoinUriBuilder, Url};
+
         let address = self
             .bitcoind()
             .map_err(|e| Error::Server(e.into()))?
             .get_new_address(None, None)
             .map_err(|e| Error::Server(e.into()))?
             .assume_checked();
-        let uri_string = if let Some(amount) = amount {
-            format!(
-                "{}?amount={}&pj={}",
-                address.to_qr_uri(),
-                amount.to_btc(),
-                self.config.pj_endpoint
-            )
-        } else {
-            format!("{}?pj={}", address.to_qr_uri(), self.config.pj_endpoint)
-        };
-        let uri = payjoin::Uri::try_from(uri_string.clone())
-            .map_err(|_| Error::Server(anyhow!("Could not parse payjoin URI string.").into()))?;
-        let _ = uri.assume_checked(); // we just got it from bitcoind above
+        let pj_endpoint =
+            Url::parse(&self.config.pj_endpoint).map_err(|e| Error::Server(e.into()))?;
+        let mut pj_uri = PayjoinUriBuilder::new(address, pj_endpoint);
+        pj_uri.amount(amount.unwrap());
```

Comment:
The presence of unwraps is a good sign that there is still error handling to complete